### PR TITLE
validate-blocked-edges: match target version in content and filename

### PR DESCRIPTION
**Pull request by petr-muller**
Prevent gotchas like the one in https://github.com/openshift/cincinnati-graph-data/pull/3243 that needed to be fixed in https://github.com/openshift/cincinnati-graph-data/pull/3249:

When run on https://github.com/openshift/cincinnati-graph-data/pull/3243, this results in the following validation failure:

```console
$ ./hack/validate-blocked-edges.py
Traceback (most recent call last):
  File "/home/pmuller/Projects/RH/cincinnati-graph-data/./hack/validate-blocked-edges.py", line 79, in <module>
    validate_blocked_edges(directory='blocked-edges')
  File "/home/pmuller/Projects/RH/cincinnati-graph-data/./hack/validate-blocked-edges.py", line 13, in validate_blocked_edges
    raise ValueError('invalid blocked edge {}: {}'.format(path, error)) from None
ValueError: invalid blocked edge blocked-edges/4.12.3-leaked-machineconfig.yaml: conditional risk to version 4.12.2 should be declared in a 4.12.2-<...>.yaml file
```

### CODE REVIEW

File: hack/validate-blocked-edges.py [hunk 1]
```diff
@@ -8,15 +8,20 @@
 def validate_blocked_edges(directory):
     for path, data in util.walk_yaml(directory=directory, allowed_extensions=('.yaml',)):
         try:
-            validate_blocked_edge(data=data)
+            validate_blocked_edge(data=data, path=path)
         except Exception as error:
-            raise ValueError('invalid blocked edge {}: {}'.format(path, error))
+            raise ValueError('invalid blocked edge {}: {}'.format(path, error)) from None
```
Comment (petr-muller) [inline hack/validate-blocked-edges.py:13]: This is unrelated to the fix, but results in less clutter. It prevents Python from writing out a `During handling of the above exception, another exception occurred` part with a second traceback for the original exception.

**Comment by petr-muller** [comment]
That's some antique Python running there...

**Comment by petr-muller** [comment]
https://github.com/openshift/release/pull/37174

**Review by wking — APPROVED** [review]
/lgtm

**Pull request merged**